modules/nf-core/metabat2/metabat2/main.nf

```diff
@@ -24,15 +24,16 @@ process METABAT2_METABAT2 {
     script:
     def args             = task.ext.args   ?: ''
     def prefix           = task.ext.prefix ?: "${meta.id}"
-    def decompress_depth = depth           ? "gzip -d -f $depth"    : ""
-    def depth_file       = depth           ? "-a ${depth.baseName}" : ""
+    def clean_depth      = depth.toString() - ".gz"
+    def decompress_depth = (depth && depth.toString() != clean_depth) ? "gzip -d -f $depth" : ""
+    def depth_input      = depth ? "-a ${clean_depth}" : ""
```

Comment:
Trying to understand this:
What happens when `depth.toString()` does not contain `.gz` (when its uncompressed) does this then just return the uncompressed filename to depth_input? 

If thats the case it looks good to go.

Reply:
Yes, that's the idea, the `-` operator removes the ".gz" substring if it's present and leaves the string untouched otherwise. That said, I dug a little more into it and apparently if the string were to contain (for some reason) ".gz" earlier in its name that would be removed preferentially - so I've switched it out to a regex that explicitly matches the end of the string.